packages/module/src/ColumnManagement/ColumnManagement.tsx

```diff
@@ -0,0 +1,164 @@
+  key: string;
+  /** The actual display name of the column possibly with a tooltip or icon. */
+  title: React.ReactNode;
+  /** If user changes checkboxes, the component will send back column array with this property altered. */
+  isShown?: boolean;
+  /** Set to false if the column should be hidden initially */
+  isShownByDefault: boolean;
+  /** The checkbox will be disabled, this is applicable to columns which should not be toggleable by user */
+  isUntoggleable?: boolean;
+}
+
+export interface ColumnProps {
+  /** Current column state */
+  columns: Column[];
+  /* Column description text */
+  description?: string;
+  /* Column title text */
+  title?: string;
+  /** Custom OUIA ID */
+  ouiaId?: string | number;
+  /** Callback when a column is selected or deselected */
+  onSelect?: (column: Column) => void;
+  /** Callback when all columns are selected or deselected */
+  onSelectAll?: (columns: Column[]) => void;
+  /** Callback when the column order changes */
+  onOrderChange?: (columns: Column[]) => void;
+  /** Callback to save the column state */
+  onSave?: (columns: Column[]) => void;
+  /** Callback to close the modal */
+  onCancel?: () => void;
+}
+
+const ColumnManagement: FunctionComponent<ColumnProps> = (
+  { columns,
+    description,
+    title,
+    ouiaId = 'Column',
+    onSelect,
+    onSelectAll,
+    onOrderChange,
+    onSave,
+    onCancel }: ColumnProps) => {
+
+  const [ currentColumns, setCurrentColumns ] = useState(
+    () => columns.map(column => ({ ...column, isShown: column.isShown ?? column.isShownByDefault, id: column.key }))
+  );
+
+  useEffect(() => {
+    setCurrentColumns(columns.map(column => ({ ...column, isShown: column.isShown ?? column.isShownByDefault, id: column.key })));
+  }, [ columns ]);
+
+  const handleChange = index => {
+    const newColumns = [ ...currentColumns ];
+    const changedColumn = { ...newColumns[index] };
+
+    changedColumn.isShown = !changedColumn.isShown;
+    newColumns[index] = changedColumn;
+
+    setCurrentColumns(newColumns);
+    onSelect?.(changedColumn);
+  };
+
+  const onDrag = (_event, newOrder) => {
+    const newColumns = newOrder.map(item => currentColumns.find(c => c.key === item.id));
+    setCurrentColumns(newColumns);
+    onOrderChange?.(newColumns);
+  };
+
+  const handleSave = () => {
+    onSave?.(currentColumns);
+  }
+
+  const handleBulkSelect = (value: BulkSelectValue) => {
+    const allSelected = value === 'all' || value === 'page';
+    handleSelectAll(allSelected);
+  };
+
+  const handleSelectAll = (select = true) => {
+    const newColumns = currentColumns.map(c => ({ ...c, isShown: c.isUntoggleable ? c.isShown : select }));
+    setCurrentColumns(newColumns);
+    onSelectAll?.(newColumns);
+  }
+
+  return (
+    <>
+      <Title headingLevel="h3">{title}</Title>
+      {description && <div style={{ paddingBottom: '1rem' }}><p>{description}</p></div>}
```

Comment:
Instead of hardcoding styles here to create the margin you'd like, i'd recommend `className='pf-v6-u-mb-md'`

or even better yet, the title and description should be left off. This could be placed in the body of a Modal or other container which can handle the title.